lib/chef/resource/aws_ebs_volume.rb

```diff
@@ -20,6 +20,10 @@ class Chef::Resource::AwsEbsVolume < Chef::Provisioning::AWSDriver::AWSResourceW
     name =~ /^vol-[a-f0-9]{8}$/ ? name : nil
   }
 
+  attribute :machine,           kind_of: String
```

Comment:
This can be `attribute :machine, kind_of [String, AwsMachine, AWS::EC2::Instance]`

To fetch the correct machine from any of the 3 provided types you use `Chef::Resource::AwsInstance.get_aws_object(new_resource.machine, resource: new_resource)` from inside your provider.

Some examples of this in other resources:
* https://github.com/chef/chef-provisioning-aws/blob/cad5c92b4a8fb3b616e45ccccc7fb7bb45af5b2d/lib/chef/provider/aws_security_group.rb#L248-L266
* https://github.com/chef/chef-provisioning-aws/blob/cad5c92b4a8fb3b616e45ccccc7fb7bb45af5b2d/lib/chef/provisioning/aws_driver/driver.rb#L603